server/game/actions/PlayUnitAction.ts

```diff
@@ -28,11 +28,12 @@ export class PlayUnitAction extends PlayCardAction {
             context.player,
             context.source,
         );
-        const effect = context.source.getOngoingEffectValues(EffectName.EntersPlayForOpponent);
-        const player = effect.length > 0 ? RelativePlayer.Opponent : RelativePlayer.Self;
+        const playForOpponentEffect = context.source.getOngoingEffectValues(EffectName.EntersPlayForOpponent);
```

Comment:
I'm working on take control right now, do you mind just adding TODO TAKE CONTROL right here so I'll remember to include this in my change